libs/render_delegate/reader.cpp

```diff
@@ -132,10 +135,55 @@ void HydraArnoldReader::Read(const std::string &filename, AtArray *overrides,
     if (arnoldRenderDelegate == 0)
         return;
 
+    UsdStageRefPtr stage = nullptr;
+    SdfLayerRefPtr rootLayer = SdfLayer::FindOrOpen(filename);
+
     HydraArnoldAPI context(arnoldRenderDelegate);
-    UsdStageRefPtr stage = UsdStage::Open(filename, UsdStage::LoadAll);
-    // TODO check that we were able to load the stage
+    if (overrides == nullptr || AiArrayGetNumElements(overrides) == 0) {
```

Comment:
this code is very similar to the code opening the stage in UsdArnoldReader. 
HydraArnoldReader and UsdArnoldReader inherits from ProceduralReader, so what about mutualising this code in a function sounding like "ProceduralReader::OpenStageWithOverrides()"  that both reader could use ?